src/orka-builder.ts

```diff
@@ -34,9 +34,9 @@ export default class OrkaBuilder {
     this.queue = [];
   }
 
-  use(m: Middleware<any> | Middleware<any>[] = []) {
+  use(m: ReturnsMiddleware<any> | ReturnsMiddleware<any>[] | Middleware<any> | Middleware<any>[] = []) {
     m = lodash.flatten([m]).filter(lodash.identity);
-    m.forEach(__ => this.middlewares.push(__));
+    m.forEach(c => this.middlewares.push(c.arguments === 1 ? (c as ReturnsMiddleware<any>)(this.config) : c));
```

Comment:
This isn't a bad idea.  `c.length === "number of arguments"`

Reply:
`c.arguments.length` correct.